### ISSUE TITLE
Exclude SVG images from media regenerate

### ISSUE THREAD

**Issue by grappler** [wp-cli/wp-cli#2160]
Would it be possible to exclude svg image files from the media regenerate command or allow the image type to be defined.

**Comment by danielbachhuber** [comment]
We could, potentially. Could you give a bit more context to the request? What did you run into, and why would it be helpful?

**Comment by grappler** [comment]
I ran `wp media regenerate` and I got the following errors for the image file `wpzoo-signete_simple.svg`

```
Warning: Illegal string offset 'file' in commands/media.php on line 237
Notice: Uninitialized string offset: 0 in commands/media.php on line 237
Warning: Illegal string offset 'file' in commands/media.php on line 238
Notice: Uninitialized string offset: 0 in commands/media.php on line 238
Warning: Couldn't regenerate thumbnails for "wpzoo-signete_simple" (ID 2217)
```

I know that WordPress core does not support SVG images but it would be great if I could hook into wp-cli and exclude svg files.

The code that I added to allow svg to be uploaded is:
```
/**
 * Allow SVG to be uploaded
 */
function wpzoo_svgs_upload_mimes( $mimes ) {

	if( current_user_can( 'administrator' ) ) {
		$mimes['svg'] = 'image/svg+xml';
	}

	return $mimes;
}
add_filter( 'upload_mimes', 'wpzoo_svgs_upload_mimes' );
```

**Comment by danielbachhuber** [comment]
Can you share `wp --info`?

The error notice you're including seems to indicate your problem was fixed in #1783

**Comment by grappler** [comment]
I was using version `0.18.0`. Updating to version `0.20.2` helped but I did not like the error message. I found a better way to handle it in my own mini SVG plugin.

```
function wpzoo_svg_generate_attachment_metadata( $metadata, $attachment_id ) {
	if ( ! empty( $metadata ) ) {
		return $metadata;
	}

	$type = get_post_mime_type( $attachment_id );
	if ( 'image/svg+xml' == $type ) {
		$att_desc = sprintf( '"%1$s" (ID %2$d).', get_the_title( $attachment_id ), $attachment_id );
		return new WP_Error( 'svg-images', sprintf( __( 'Thumbnails not generated for SVG file: %1$s', 'wpzoo-svg' ), $att_desc ) );
	}

	return $metadata;
}
add_filter( 'wp_generate_attachment_metadata', 'wpzoo_svg_generate_attachment_metadata', 10, 2 );
```
